src/route/user-route.ts

```diff
@@ -0,0 +1,41 @@
+import { Router } from 'express';
+
+// eslint-disable-next-line import/no-cycle
+import { logger } from '../app';
+import { User } from '../entity/user';
+import { userToUserDTO } from '../dtos/user-dto';
+
+export const userRoute = Router();
+
+userRoute.get('/:id', async (req, res) => {
+    if (!req.params.id) {
+        logger.error(`Reequest parameter id is missing`);
+        res.status(400).send();
+        return;
+    }
+    const userID = req.params.id;
+    const user = await User.findOneBy({ id: userID });
+    if (!user) {
+        logger.error(`User with id ${req.params.id} not found`);
+        res.status(404).send();
+        return;
+    }
+    const userDto = userToUserDTO(user);
+    res.json(userDto);
+});
+
+userRoute.post('/', async (req, res) => {
+    const user = new User();
+    user.oidcId = req.body.oidcId;
+    user.provider = req.body.provider;
+    user.name = req.body.name;
+    user.email = req.body.email;
+    user.profile = req.body.profile;
+    let usr = await User.findOneBy({ oidcId: user.oidcId });
+    if (usr === null || usr === undefined) {
+        usr = await user.save();
+    }
+    const userDto = userToUserDTO(usr);
+    res.status(201).send();
```

Comment:
> We probably need a specific endpoint for handling OIDC logins.

...so that we can extract the context we supplied at the beginning of the login journey (i.e. before we redirected them to the IDP) and then redirect them back to the part of our site that they started on.